src/app/components/result-box/result-box.component.html

```diff
@@ -1,5 +1,5 @@
 <div class="container w-50">
-  <div class="result-box">
+  <div class="result-box" [ngClass]="(borderResultStyles)?'border-success':'border-danger'">
```

Comment:
```suggestion
  <div class="result-box" [ngClass]="borderStyles()">
```
Define a function, try to remove logic from the HTML